LSP4: Add token types (#250)
lukso-network/LIPs: LSPs/LSP-4-DigitalAsset-Metadata.md

}
```

#### LSP4TokenType

> A string representing the type of token that this contract represents.

Reviewer: Suggest putting the note above like this

> **Note:** more token types COULD be added later.

NOTE: More token types COULD be added later.

_Requirements_

This MUST NOT be changeable, and set only during initialization of the token contract.

| Value |   Type    | Description                                                                                                                                       |
| :---: | :-------: | :------------------------------------------------------------------------------------------------------------------------------------------------ |

> |  `0`  | `Token` | Only valid for LSP7, meaning its a generic token, where the `LSP4Metadata` represents the token information. |

Reviewer: Maybe it would be good to specify this extra info?

```suggestion
|  `0`  | `Token` | Only valid for LSP7 (either divisible or not), meaning its a generic token, where the `LSP4Metadata` represents the token information. |
```

> |  `1`  | `NFT`  | If the contract is LSP7 or LSP8, then the `LSP4Metadata` represents the information of a **single** NFT item, that has multiple ownable amounts or IDs. If its an LSP7 contract, the `decimals` function SHOULD to return `0`. |

Reviewer: Is this only for non divisible tokens?

```suggestion
|  `1`  | `NFT`  | If the contract is LSP7 and not divisible (`decimals` set to `0`), or LSP8, then the `LSP4Metadata` represents the information of a **single** NFT item, that has multiple ownable amounts or IDs. |
```

> |  `2`  | `Collection` | Only valid for LSP7. The `LSP4Metadata` represents the information of a the collection, and each individual token ID, requires its metadata depending on the `LSP8TokenIdType` and `LSP8MetadataTokenURI`. [See LSP8 for details](./LSP-8-IdentifiableDigitalAsset.md).                 |

Reviewer: I am not sure about this section is it:

- only for LSP7?
- only for LSP8?
- applicable to both?

> <!-- - `LSP26NFT` -->

Reviewer: Is this comment still applicable?

Author: yes, lets leave it there for now
